ncollide_geometry/query/point_internal/point_triangle.rs

```diff
@@ -18,6 +18,24 @@ fn compute_result<P: Point>(pt: &P, proj: P) -> PointProjection<P> {
 impl<P: Point, M: Isometry<P>> PointQuery<P, M> for Triangle<P> {
     #[inline]
     fn project_point(&self, m: &M, pt: &P, solid: bool) -> PointProjection<P> {
+        let (projection, _) = self.project_point_with_extra_info(m, pt, solid);
+        projection
+    }
+
+    // NOTE: the default implementation of `.distance_to_point(...)` will return the error that was
+    // eaten by the `::approx_eq(...)` on `project_point(...)`.
+}
+
+impl<P: Point, M: Isometry<P>> RichPointQuery<P, M> for Triangle<P> {
+    // Implementing this trait while providing no projection info might seem
+    // nonsensical, but it actually makes it possible to complete the
+    // `RichPointQuery` implementation for `BaseMesh`.
+    type ExtraInfo = ();
```

Comment:
I looked into adding that myself, but decided against it. I'm not at home right now and don't have access to my own copy of Real-Time Collision Detection, and as there don't seem to be any unit tests, I was afraid I'd mess it up in subtle ways.